deeplake/api/dataset.py

```diff
@@ -999,6 +1000,87 @@ def connect(
+                - a local file system path of the form ``./path/to/dataset`` or ``~/path/to/dataset`` or ``path/to/dataset``.
+                - a memory path of the form ``mem://path/to/dataset`` which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            key_to_tensor_mapping (Optional[Dict]): A one-to-one mapping between COCO keys and Dataset tensor names.
+            file_to_group_mapping (Optional[Dict]): A one-to-one mapping between COCO annotation file names and Dataset group names.
+            ignore_one_group (bool): Skip creation of group in case of a single annotation file. Set to ``True`` by default.
+            ignore_keys (List[str]): A list of COCO keys to ignore.
+            image_settings (Optional[Dict]): A dictionary containing settings for the images tensor.
+            src_creds (Optional[Dict]): Credentials to access the source path. If not provided, will be inferred from the environment.
+            dest_creds (Optional[Dict]): A dictionary containing credentials used to access the destination path of the dataset.
+            inspect_limit (int): The maximum number of samples to inspect in the annotations json. Set to ``1000000`` by default.
+            progressbar (bool): Enables or disables ingestion progress bar. Set to ``True`` by default.
+            num_workers (int): The number of workers to use for ingestion. Set to ``0`` by default.
+            **dataset_kwargs: Any arguments passed here will be forwarded to the dataset creator function. See :func:`deeplake.dataset`.
+
+        Returns:
+            Dataset: The Dataset created from images and COCO annotations.
+
+        Raises:
+            IngestionError: If either ``key_to_tensor_mapping`` or ``file_to_group_mapping`` are not one-to-one.
+        """
+        dest = convert_pathlib_to_string_if_needed(dest)
+        images_directory = convert_pathlib_to_string_if_needed(images_directory)
+        annotation_files = (
+            [convert_pathlib_to_string_if_needed(f) for f in annotation_files]
+            if isinstance(annotation_files, list)
+            else convert_pathlib_to_string_if_needed(annotation_files)
+        )
+
+        if isinstance(dest, Dataset):
+            ds = dest
+        else:
+            dest = convert_pathlib_to_string_if_needed(dest)
+            ds = deeplake.dataset(dest, creds=dest_creds, **dataset_kwargs)
+
+        unstructured = CocoDataset(
+            source=images_directory,
+            annotation_files=annotation_files,
+            key_to_tensor_mapping=key_to_tensor_mapping,
+            file_to_group_mapping=file_to_group_mapping,
+            ignore_one_group=ignore_one_group,
+            ignore_keys=ignore_keys,
+            image_settings=image_settings,
+            creds=src_creds,
+        )
+        structure = unstructured.prepare_structure(inspect_limit)
+        structure.create_missing(ds)
```

Comment:
do we want to check whether dataset is empty/has tensors of equal length before stating ingestion?

Reply:
With new API, only empty datasets are allowed, so apparently there is no need for now